aws/resource_aws_codedeploy_deployment_config.go

```diff
@@ -137,15 +226,102 @@ func expandAwsCodeDeployConfigMinimumHealthHosts(d *schema.ResourceData) *codede
 	return &minimumHealthyHost
 }
 
+func expandAwsCodeDeployTrafficRoutingConfig(d *schema.ResourceData) *codedeploy.TrafficRoutingConfig {
+	block, ok := d.GetOk("traffic_routing_config")
+	if !ok {
+		return nil
+	}
+	config := block.([]interface{})[0].(map[string]interface{})
+	trafficRoutingConfig := codedeploy.TrafficRoutingConfig{}
+
+	if trafficType, ok := config["type"]; ok {
+		trafficRoutingConfig.SetType(trafficType.(string))
```

Comment:
For consistency with most of the provider codebase, we should directly set the attributes instead of calling the setter methods, here and below, e.g.

```suggestion
		trafficRoutingConfig.Type = aws.String(trafficType.(string))
```